mesonbuild/dependencies/base.py

```diff
@@ -46,6 +46,19 @@ class DependencyException(MesonException):
     '''Exceptions raised while trying to find dependencies'''
 
 
+class MissingCompiler:
+    """Represent a None Compiler - when no tool chain is found.
+    replacing AttributeError with DependencyException"""
+
+    def __getattr__(self, item):
```

Comment:
```suggestion
    def __getattr__(self, item: str) -> T.Any:
```

I don't know what mypy possibly expects here for the return type???

Hmm, perhaps:

```suggestion
    def __getattr__(self, item: str) -> T.NoReturn:
```

Reply:
NoReturn is not callable, we should just use Any here I guess. :)

Reply:
done